### PR TITLE
Fix copyright lines with no authors and add tests

### PR DESCRIPTION
**Pull request by Rapiz1**
Fix copyright lines with no authors and add tests
spins off https://github.com/chapel-lang/chapel/pull/15137#issuecomment-596859912
@lydia-duncan Please review this.

### CODE REVIEW

File: test/chpldoc/compflags/author/withoutAuthor.doc.prediff [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+grep Copyright $2 > $2.prediff.tmp
```
Comment (lydia-duncan) [inline test/chpldoc/compflags/author/withoutAuthor.doc.prediff]: Note that because you will have multiple tests in this directory by the time both PRs are merged, and you are using the default docs location, this may pick up the results of other runs if something happens to go wrong with the current test.  I would recommend setting a different documentation output directory and *only* checking that directory for your result.

Otherwise, this change looks right to me!

File: third-party/chpl-venv/chpldoc-sphinx-project/source/conf.py [hunk 2]
```diff
@@ -53,7 +53,10 @@
 
 author_text = os.environ.get('CHPLDOC_AUTHOR', 'AUTHOR TEXT')
 
-copyright = u'2015, {0}'.format(author_text)
+if len(author_text):
+    copyright = u'2015, {0}'.format(author_text)
+else:
+    copyright = u'2015'
```
Comment (lydia-duncan) [inline third-party/chpl-venv/chpldoc-sphinx-project/source/conf.py]: Except maybe we should be smarter about choosing a copyright year (but that'd be good to handle in another PR, probably)

Comment (Rapiz1) [reply third-party/chpl-venv/chpldoc-sphinx-project/source/conf.py]: I actually don't know about the stuff like how to choose a copyright year and  what it means... So I just left it there.

File: test/chpldoc/compflags/author/withoutAuthor.doc.prediff (hunk 1 above)
Comment (Rapiz1) [reply test/chpldoc/compflags/author/withoutAuthor.doc.prediff]: What about putting them in separated subdirs? I think it will be more clear.

Comment (lydia-duncan) [reply test/chpldoc/compflags/author/withoutAuthor.doc.prediff]: I don't have any problems with that, but grep should still only search the generated docs directory

Comment (Rapiz1) [reply test/chpldoc/compflags/author/withoutAuthor.doc.prediff]: I just think with catfiles, only the docs/modules/withoutAuthor.doc.html will be passed to grep and it should be fine. Have I missed something?

Comment (lydia-duncan) [reply test/chpldoc/compflags/author/withoutAuthor.doc.prediff]: Oh dur, you're right!  I misread the command, sorry about that

Comment (lydia-duncan) [reply test/chpldoc/compflags/author/withoutAuthor.doc.prediff]: I guess no changes are needed then, lemme run this through testing :)

File: test/chpldoc/compflags/author/withoutAuthor/CLEANFILES [hunk 3]
```diff
@@ -0,0 +1 @@
+docs
```
Comment (lydia-duncan) [inline test/chpldoc/compflags/author/withoutAuthor/CLEANFILES]: Whoops, missing newline at end of file here

**Comment by lydia-duncan** [comment]
This passed a full paratest run.  I need to look at the generated documentation still (not sure if I will have time today, feeling more release pressure at the moment)

File: test/chpldoc/compflags/author/withoutAuthor/CLEANFILES (hunk 3 above)
Comment (Rapiz1) [reply test/chpldoc/compflags/author/withoutAuthor/CLEANFILES]: Done

**Comment by lydia-duncan** [comment]
In building the generated documentation, I ran the test on a Mac instead of a Linux machine and am getting different behavior.  This is the error message I got:

```
[Starting /Users/lydia/Documents/Repositories/chapel/util/test/sub_test Thu Mar 12 15:43:15 PDT 2020]
[Starting subtest - Thu Mar 12 15:43:15 PDT 2020]
[test: chpldoc/compflags/author/withoutAuthor/withoutAuthor.doc.chpl]
[Executing compiler /Users/lydia/Documents/Repositories/chapel/bin/darwin-x86_64/chpldoc withoutAuthor.doc.chpl < /dev/null]
[Elapsed compilation time for "chpldoc/compflags/author/withoutAuthor/withoutAuthor.doc" - 1.187 seconds]
[Concatenating extra files: withoutAuthor.doc.catfiles]
[Executing prediff withoutAuthor.doc.prediff]
[Executing diff withoutAuthor.doc.good withoutAuthor.doc.comp.out.tmp]
1c1
<         &copy; Copyright n
---
>         &copy; Copyright 2015
[Error matching compiler output for chpldoc/compflags/author/withoutAuthor/withoutAuthor.doc]
[Elapsed time to compile and execute all versions of "chpldoc/compflags/author/withoutAuthor/withoutAuthor.doc" - 1.280 seconds]
[Finished subtest "chpldoc/compflags/author/withoutAuthor" - 1.646 seconds]
```

Maybe there's something different about sed on Macs versus Linux machines?  Lemme know if you need me to run stuff for you in environments you don't have access to

**Comment by Rapiz1** [comment]
I think it should work now.

**Comment by lydia-duncan** [comment]
The test now works on Mac and Linux!  I looked at the docs built by the test run and our generated online documentation with this change, and the copyright in both places looks good.

I will merge this change on Monday to avoid checking testing over the weekend, thanks for your hard work and patience!

**Comment by Rapiz1** [comment]
You're welcome. I'm happy to contribute to Chapel :)

**Comment by bradcray** [comment]
Hi @Rapiz1 and @lydia-duncan: Unfortunately, this test is failing in some configurations with:

```chapel
[Executing prediff withoutAuthor.doc.prediff]
sed: invalid option -- 'E'
```

This seems familiar, as though many prediff scripts have made this mistake and then worked around it.  Maybe check what other prediff scripts do to resolve it?

**Comment by Rapiz1** [comment]
@bradcray Thanks for your testing! I find that there's no way to use ERE in `sed` with an identical option on different systems. Current script will fail with old `sed`. However, I managed to work around by using BRE. 
A PR is opened. https://github.com/chapel-lang/chapel/pull/15248
